src/styles/getTooltipStyles.js

```diff
@@ -174,6 +179,27 @@ export default function getTooltipStyles(
         //   3) Add the horizontal shift (left or right) computed above to keep it out of the gutters.
         //   4) Lastly, add the manual horizontal shift passed in as a parameter.
         wrapperLeft = xOffset + (componentWidth / 2 - tooltipWidth / 2) + horizontalShift + manualShiftHorizontal;
+
+        // By default, the pointer's top-left will align with the top-left of the tooltip wrapper.
+        //
+        // To align it vertically, we'll:
+        //   If the pointer should be below the tooltip wrapper, shift the pointer down (+) by the tooltip height,
+        //   so that the top of the pointer lines up with the bottom of the tooltip
+        //
+        //   OR if the pointer should be above the tooltip wrapper, then the pointer up (-) by the pointer's height
+        //   so that the bottom of the pointer lines up with the top of the tooltip
+        pointerWrapperTop = shouldShowBelow ? -POINTER_HEIGHT : tooltipHeight;
+
+        // To align it horizontally, we'll:
+        //   1) Shift the pointer to the right (+) by the half the tooltipWidth's width,
+        //      so the left edge of the pointer lines up with the tooltipWidth's center.
+        //   2) To the left (-) by half the pointer's width,
+        //      so the pointer's center lines up with the tooltipWidth's center.
+        //   3) Due to the tip start from the left edge of wrapper Tooltip so we have to remove the
+        //      horizontalShift which is added to adjust it into the Window
```

Comment:
I see you didn't write the original comment here, but I'm not quite sure what this third point is trying to say. I've read it a few times and I'm not sure I understand it. Is there a better way of explaining this?

Reply:
I also don't get it by reading the comment, but if I can guess, we want to shift horizontally the arrow (pointer) to the center of the hovered component in case the tooltip box is shifted horizontally to prevent it's going out of window.

For example, if we remove the horizontalShfitPointer, we will see this

<img width="333" alt="image" src="https://github.com/Expensify/App/assets/50919443/2399b5d9-f4a2-4a70-a4dc-7ad5be0bac0c">

The arrow is at the center of the tooltip, but not at the center of the component I hover (the +4 avatar). That is because the tooltip box is shifted horizontally to the right by the amount of `horizontalShift` to prevent it's going out of window. To make the arrow back to the center of the hovered component, it's basically negated by the amount of the `horizontalShift`. And this is how it supposed to look
<img width="364" alt="image" src="https://github.com/Expensify/App/assets/50919443/45bc5035-92a0-437c-a07a-8b69a07ee4e6">

I guess we can write something like:
```
Remove the wrapper's horizontalShift to maintain the pointer at the center of the hovered component.
```

What do you think?

Reply:
Yeah that comment sounds good to me 👍

Reply:
Updated